wapitiCore/attack/mod_wapp.py

```diff
@@ -106,30 +146,75 @@ def __init__(self, crawler, persister, attack_options, stop_event, crawler_confi
         if not os.path.isdir(self.user_config_dir):
             os.makedirs(self.user_config_dir)
 
+    async def copy_files_to_conf(self, file1: str, file2: str, file3: str):
+        """
+        This function copies wapp DB files specified as arguments to the config directory.
+        """
+        files_to_copy = [file1, file2, file3]
+
+        for source_file in files_to_copy:
+            # Check if file exists before attempting to copy
+            if not os.path.isfile(source_file):
+                logging.error(f"Warning: File {source_file} does not exist, skipping.")
+                continue
+
+            # Construct the destination file path using the filename
+            destination_file = os.path.join(self.user_config_dir, os.path.basename(source_file))
+
+            try:
+                shutil.copy(source_file, destination_file)
+                logging.info(f"Copied {source_file} to {destination_file}")
+            except shutil.Error as err:
+                logging.error(f"Error copying {source_file}: {err}")
+
     async def update(self):
         """Update the Wappalizer database from the web and load the patterns."""
 
         wapp_categories_url = f"{self.BASE_URL}src/categories.json"
         wapp_technologies_base_url = f"{self.BASE_URL}src/technologies/"
         wapp_groups_url = f"{self.BASE_URL}src/groups.json"
-        if not is_valid_url(self.BASE_URL):
-            raise InvalidOptionValue(
-                "--wapp-url", self.BASE_URL
-            )
-        try:
-            await self._load_wapp_database(
-                wapp_categories_url,
-                wapp_technologies_base_url,
-                wapp_groups_url
-            )
-        except RequestError:
-            self.network_errors += 1
-            raise
-        except IOError:
-            logging.error("Error downloading wapp database.")
-        except ValueError as e:
-            logging.error(f"Value error: {e}")
-            raise
+        if self.WAPP_DIR:
+            categories_file_path = os.path.join(self.WAPP_DIR, self.WAPP_CATEGORIES)
+            groups_file_path = os.path.join(self.WAPP_DIR, self.WAPP_GROUPS)
+            technologies_directory_path = os.path.join(self.WAPP_DIR, "technologies/")
+            technologies_file_path = os.path.join(self.WAPP_DIR, self.WAPP_TECHNOLOGIES)
+            try:
+                merge_json_files(technologies_directory_path, technologies_file_path)
+            except (ValueError, OSError) as error:
+                logging.error(error)
+                raise ValueError(f"Update failed : Something went wrong with files in {self.WAPP_DIR}") from error
+            try:
+                if is_json_file(categories_file_path) and is_json_file(groups_file_path) \
+                        and is_json_file(technologies_file_path):
+                    try:
+                        await self.copy_files_to_conf(categories_file_path, technologies_file_path, groups_file_path)
```

Comment:
Use list for files : `[categories_file_path, technologies_file_path, groups_file_path]`